hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMDBCheckpointServletInodeBasedXfer.java

```diff
@@ -0,0 +1,386 @@
+        .getConfiguration();
+  }
+
+  /**
+   * Collects paths to all snapshot databases.
+   *
+   * @param omMetadataManager OMMetadataManager instance
+   * @return Set of paths to snapshot databases
+   * @throws IOException if an I/O error occurs
+   */
+  Set<Path> getSnapshotDirs(OMMetadataManager omMetadataManager) throws IOException {
+    Set<Path> snapshotPaths = new HashSet<>();
+    SnapshotChainManager snapshotChainManager = new SnapshotChainManager(omMetadataManager);
+    for (SnapshotChainInfo snapInfo : snapshotChainManager.getGlobalSnapshotChain().values()) {
+      String snapshotDir =
+          OmSnapshotManager.getSnapshotPath(getConf(), SnapshotInfo.getCheckpointDirName(snapInfo.getSnapshotId()));
+      Path path = Paths.get(snapshotDir);
+      snapshotPaths.add(path);
+    }
+    return snapshotPaths;
+  }
+
+  /**
+   * Writes database files to the archive, handling deduplication based on inode IDs.
+   *
+   * @param sstFilesToExclude Set of SST file IDs to exclude from the archive
+   * @param dbDir Directory containing database files to archive
+   * @param maxTotalSstSize Maximum total size of SST files to include
+   * @param archiveOutputStream Archive output stream
+   * @param tmpDir Temporary directory for processing
+   * @return true if processing should continue, false if size limit reached
+   * @throws IOException if an I/O error occurs
+   */
+  private boolean writeDBToArchive(Set<String> sstFilesToExclude, Path dbDir, AtomicLong maxTotalSstSize,
+      ArchiveOutputStream<TarArchiveEntry> archiveOutputStream, Path tmpDir,
+      Map<String, String> hardLinkFileMap, Path destDir) throws IOException {
+    if (!Files.exists(dbDir)) {
+      LOG.warn("DB directory {} does not exist. Skipping.", dbDir);
+      return true;
+    }
+    try (Stream<Path> files = Files.list(dbDir)) {
+      Iterable<Path> iterable = files::iterator;
+      for (Path dbFile : iterable) {
+        if (!Files.isDirectory(dbFile)) {
+          String fileId = OmSnapshotUtils.getInodeAndMtime(dbFile);
+          String path = dbFile.toFile().getAbsolutePath();
+          if (destDir != null) {
+            path = destDir.resolve(dbFile.getFileName()).toString();
+          }
+          // if the file is in the om checkpoint dir, then we need to change the path to point to the OM DB.
+          if (path.contains(OM_CHECKPOINT_DIR)) {
+            path = getDbStore().getDbLocation().toPath().resolve(dbFile.getFileName()).toAbsolutePath().toString();
+          }
+          hardLinkFileMap.put(path, fileId);
+          if (!sstFilesToExclude.contains(fileId)) {
+            long fileSize = Files.size(dbFile);
+            if (maxTotalSstSize.get() - fileSize <= 0) {
+              return false;
+            }
+            linkAndIncludeFile(dbFile.toFile(), fileId, archiveOutputStream, tmpDir);
+            maxTotalSstSize.addAndGet(-fileSize);
+            sstFilesToExclude.add(fileId);
```

Comment:
Could you also log the progress of the tarball transfer like in 
https://github.com/apache/ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMDBCheckpointServlet.java#L621-L624

Reply:
Done.